### DISCUSSION TITLE
Postprocessing consequences

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3800] [Q&A] [answered]
Once I enabled post-processing on my view, I noticed some changes.  For one thing, the background, which was previously set to white via

```
_renderer->setClearOptions({.clearColor={1.0, 1.0, 1.0, 1.0}, .clear = true});
```

became gray.  Second, I had to greatly reduce the intensity of indirect light to get a similar appearance to what I had before.  Are these changes to be expected?  I'm guessing that they might be related to tone mapping, but there does not seem to be any way to turn off tone mapping without turning off post-processing.

**Comment by romainguy** [comment, marked as answer]
You can turn off tone mapping by using the linear tone mapper (see the ColorGrading API). Of course this means that HDR lighting will clip.

I'm not sure what you mean about the intensity but tone mapping applies compression to lighting effectively. So if you disabled tone mapping your lighting might appear harsh or clipped (see above).

**Reply by jwwalker** [reply]
Maybe some images will help.  First, some test objects with no post-processing:

![no-post-processing](https://user-images.githubusercontent.com/730426/114645394-3bf2f780-9c8e-11eb-9b54-56db1a330bc6.png)

Then the same scene with LINEAR tone-mapping:

![linear tonemapping](https://user-images.githubusercontent.com/730426/114645442-5cbb4d00-9c8e-11eb-8121-73cdc566a4cc.png)

And finally the same scene with ACES tone-mapping.

![ACES tonemapping](https://user-images.githubusercontent.com/730426/114645469-693fa580-9c8e-11eb-892c-d42900c66852.png)

In the LINEAR case, the background is nearly white, but the objects are much paler than in the no-post-processing case.  The light intensities and camera exposure are identical in these 3 cases.

**Reply by romainguy** [reply]
Tone mapping — and color grading in general — is applied via a 3D LUT so there will be approximations. You can use the quality setting of the `ColorGrading` API to improve this or, better yet, apply an inverse tone map function to your background (in the case of white you could use values greater than 1.0).

The main difference you are seeing between post processing off vs post processing on is that post processing off does **not** properly apply the sRGB OETF (opto-electronic transfer function) on the output, which is very much incorrect since the display will apply its EOTF and produce a result that looks overly dark/contrasty like what you are observing.

**Reply by jwwalker** [reply]
Thanks.  With linear tone mapping, setting the clear color to 1.1 suffices to make the background white.  With ACES, the clear color would need to be increased so much that the pink translucent ball totally disappears.

**Reply by romainguy** [reply]
With ACES, HDR value ~8.0 maps to 1.0 in SDR.
